### PR TITLE
Bug 1881210: desiredNodePortService: Add port 1936

### PR DESCRIPTION
**Pull request by Miciah**
#### `desiredRouterDeployment`: Fix comment on host ports

* `pkg/operator/controller/ingress/deployment.go` (`desiredRouterDeployment`): Mention that the deployment exposes port 1936 in addition to ports 80 and 443.

#### `desiredNodePortService`: Add port 1936

When creating the `NodePort`-type service for an ingresscontroller that specifies the `NodePortService` endpoint publishing strategy type, include a "metrics" port with port 1936 so that external load balancers can use the `/healthz/ready` endpoint.

* `pkg/operator/controller/ingress/nodeport_service.go` (`ensureNodePortService`): Pass an argument to `desiredNodePortService` to tell it not to add a "metrics" port if the service already exists and omits the port.
(`desiredNodePortService`): Add a `wantMetricsPort` parameter.  Add port 1936 to the service's ports spec if `wantMetricsPort` is true.
* `pkg/operator/controller/ingress/nodeport_service_test.go` (`TestDesiredNodePortService`): Add a test case where `wantMetricsPort` is true.  Verify that `desiredNodePortService` returns the expected service. (`TestNodePortServiceChanged`): Update the service used for tests to be consistent with what `desiredNodePortService` returns.
* `test/e2e/operator_test.go` (`TestNodePortServiceEndpointPublishingStrategy`): Verify that the operator creates the `NodePort`-type service with the expected ports.  Verify that the operator does not add the "metrics" port if it is is removed from the service.

### CODE REVIEW

**Comment by sgreene570** [comment]
@Miciah do we want to revive this PR?

File: pkg/operator/controller/ingress/nodeport_service.go [hunk 1]
```diff
@@ -24,13 +24,26 @@ import (
...
 func (r *reconciler) ensureNodePortService(ic *operatorv1.IngressController, deploymentRef metav1.OwnerReference) (bool, *corev1.Service, error) {
-	wantService, desired := desiredNodePortService(ic, deploymentRef)
-
 	haveService, current, err := r.currentNodePortService(ic)
 	if err != nil {
 		return false, nil, err
 	}
 
+	// For compatibility, omit the "metrics" port iff the service already
+	// exists and doesn't have a "metrics" port.
```
Comment (sgreene570) [inline pkg/operator/controller/ingress/nodeport_service.go:33]: Could you add a comment here mentioning that this logic allows a user to remove the metrics port from the node port service (as discussed offline) ?

**Comment by sgreene570** [comment]
> * `test/e2e/operator_test.go` (`TestNodePortServiceEndpointPublishingStrategy`): Verify that the operator creates the `NodePort`-type service with the expected ports.  Verify that the operator does not add the "metrics" port of it is is removed from the service.

Small commit message typo. "of it is removed" -> "if it is removed".

File: pkg/operator/controller/ingress/nodeport_service.go (hunk 1 above)
Comment (Miciah) [reply pkg/operator/controller/ingress/nodeport_service.go]: Added.  Thanks!

**Comment by Miciah** [comment]
Updated per Steve's comments.

**Pull request merged**